File: 05-telnyx-media-streaming/node/create-agent.ts
import { PhonicClient } from "phonic";
import { phonicApiKey } from "./env-vars";

const client = new PhonicClient({
  apiKey: phonicApiKey,
});

async function createAgent() {
  await client.agents.create({
    name: "agent-telnyx",
    welcome_message: "Hi there. How can I help you today?",
    system_prompt: "Be helpful, friendly, and concise.",
    // mulaw_8000 == PCMU 8 kHz, which is what Telnyx streams by default.
    audio_format: "mulaw_8000",
  });
}

createAgent();

File: 05-telnyx-media-streaming/python/create_agent.py
import os

from dotenv import load_dotenv
from phonic import Phonic

load_dotenv(".env.local")

client = Phonic(
    api_key=os.getenv("PHONIC_API_KEY"),
)

client.agents.create(
    name="agent-telnyx",
    welcome_message="Hi there. How can I help you today?",
    system_prompt="Be helpful, friendly, and concise.",
    # mulaw_8000 == PCMU 8 kHz, which is what Telnyx streams by default.
    audio_format="mulaw_8000",
)
